objmodel/chapter.md

```diff
@@ -885,44 +877,44 @@ class Instance(Base):
             self.map = new_map
 ````
 
-The class now passes ``None`` as the fields dict to ``Base``, as ``Instance``
-will now store the content of the dict in another way. Therefore it now needs
-to override the the ``_read_dict`` and ``_write_dict`` methods. In a real
-implementation, we would refactor the ``Base`` class to not be responsible for
-storing the field dictionary any more, but for this section having instances
+The class now passes ``None`` as the fields dictionary to ``Base``, as ``Instance``
+will store the content of the dictionary in another way. Therefore it needs
+to override the ``_read_dict`` and ``_write_dict`` methods. In a real
+implementation, we would refactor the ``Base`` class so that it is no longer responsible for
+storing the fields dictionary, but for now having instances
 store ``None`` there is good enough.
 
-When creating a new instance it starts out with using the ``EMPTY_MAP``, which
-has no attributes, and an empty storage. To implement ``_read_dict``, the
+When creating a new instance, the class starts out using the ``EMPTY_MAP``, which
+has no attributes, and empty storage. To implement ``_read_dict``, the
 instance's map is asked for the index of the attribute name. Then the
 corresponding entry of the storage list is returned.
 
 Writing into the fields dictionary has two cases. On the one hand the value of
 an existing
 attribute can be changed. This is done by simply changing the storage at the
-corresponding index. If the attribute does not exist yet, a *map transition* is
+corresponding index. On the other hand, if the attribute does not exist yet, a *map transition* is
 needed using the ``next_map`` method. The value of the new attribute is appended
 to the storage list.
 
 
-What does this optimization achieve? It optimizes the memory use of the common
-case where many instances that have the same layout exist. It is not a universal
-optimization: Code that creates instances with wildly different sets of
-attributes will have a larger memory footprint than just using dictionaries.
+What does this optimization achieve? It optimizes use of memory in the common
+case where there are many instances with the same layout. It is not a universal
+optimization: code that creates instances with wildly different sets of
+attributes will have a larger memory footprint than if we just use dictionaries.
 
-This is a common approach for optimizing dynamic languages. It is often not
-possible to find optimizations that are faster or use less memory in *all*
-cases. To still speed up programs in practice, optimizations are chosen that
-should apply to how the language is typically used in practice, while potentially
-making the behaviour worse for programs that use extremely dynamic features.
+This is a common problem when optimizing dynamic languages. It is often not
+possible to find optimizations that are faster or use less memory in all
+cases. In practice, optimizations are made that
```

Comment:
I think I prefer "chosen" instead of "made" here, to really make clear that it's a conscious choice by the language implementer, who picks optimizations to use out of a (usually infinite) set of possible ones.